src/libs/auth.ts

```diff
@@ -97,71 +127,206 @@ const revokeTokens = async () => {
+          }));
+        }
+        authStore.update((state) => ({
+          ...state,
+          authTokenMetadata: {
+            ...authStore.get().authTokenMetadata,
+            token: user.access_token,
+            createdAt: authTokenCreatedAt,
+            expiresAt: authTokenExpiresAt,
+            id: authTokenId,
+            // TODO: Change this to only update if the auth token is different than the previous token
+            totalTokensUsedThisSession: authStore.get().authTokenMetadata.totalTokensUsedThisSession + 1,
+          },
+        }));
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.success, {
+          authProvider: (user.profile as B2cIdTokenClaims).idp,
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          authTokenCreatedAt: labelTimestampMetric(authTokenCreatedAt),
+          authTokenExpiresAt: labelTimestampMetric(authTokenExpiresAt),
+          authTokenId,
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+          jwtExpiresAt: labelTimestampMetric(jwtExpiresAt),
+        });
+      },
+    ],
+    [
+      'expired',
+      () => {
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.expired, {
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+        });
+      },
+    ],
+    [
+      'error',
+      () => {
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.error, {
+          // we could potentially log the reason, but I don't know if that data is safe to log
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+        });
+      },
+    ]
+  );
+
+  return loadedAuthTokenState;
+};
+
+const tryLoadAuthToken = async (includeBillingScope = false, popUp = false): Promise<AuthTokenState> => {
+  const args: ExtraSigninRequestArgs = getSigninArgs(includeBillingScope);
+  const authInstance = getAuthInstance();
+  const loadedAuthTokenResponse: User | null | AuthTokenErrorState = await (popUp
+    ? // returns Promise<User | null>, attempts to use the refresh token to get a new authToken
+      authInstance.signinPopup(args)
+    : authInstance.signinSilent(args)
+  ).catch((e) => {
+    const errorMsg = 'An unexpected exception occurred while attempting to load auth credentials.';
```

Comment:
Since it is not our function, I am not sure what is getting sent back in this error block. I think once we have more time to gather metrics on authTokenErrors, we may be able to give a more clear message here.